test/unit/org/apache/cassandra/db/virtual/GossipInfoTableTest.java

```diff
@@ -33,64 +36,58 @@
 import org.apache.cassandra.gms.Gossiper;
 import org.apache.cassandra.locator.InetAddressAndPort;
 
+import static com.google.common.collect.ImmutableList.of;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class GossipInfoTableTest extends CQLTester
 {
-    private static final String KS_NAME = "vts";
-
-    @SuppressWarnings("FieldCanBeLocal")
-    private GossipInfoTable table;
-
     @BeforeClass
     public static void setUpClass()
     {
         CQLTester.setUpClass();
     }
 
-    @Before
-    public void config()
-    {
-        table = new GossipInfoTable(KS_NAME);
-        VirtualKeyspaceRegistry.instance.register(new VirtualKeyspace(KS_NAME, ImmutableList.of(table)));
-    }
-
     @Test
     public void testSelectAllWhenGossipInfoIsEmpty() throws Throwable
     {
-        assertEmpty(execute("SELECT * FROM vts.gossip_info"));
+        // we have not triggered gossiper yet
+        VirtualKeyspaceRegistry.instance.register(new VirtualKeyspace("vts_1",
+                                                                      of(new GossipInfoTable("vts_1", HashMap::new))));
+        assertEmpty(execute("SELECT * FROM vts_1.gossip_info"));
     }
 
-    @SuppressWarnings("deprecation")
     @Test
     public void testSelectAllWithStateTransitions() throws Throwable
     {
         try
         {
             requireNetwork(); // triggers gossiper
 
-            UntypedResultSet resultSet = execute("SELECT * FROM vts.gossip_info");
+            ConcurrentMap<InetAddressAndPort, EndpointState> states = Gossiper.instance.endpointStateMap;
+            Awaitility.await().until(() -> !states.isEmpty());
+            Map.Entry<InetAddressAndPort, EndpointState> entry = states.entrySet().stream().findFirst()
+                    .orElseThrow(AssertionError::new);
+            InetAddressAndPort endpoint = entry.getKey();
+            EndpointState localState = new EndpointState(entry.getValue());
 
-            assertThat(resultSet.size()).isEqualTo(1);
-            assertThat(Gossiper.instance.endpointStateMap.size()).isEqualTo(1);
+            Supplier<Map<InetAddressAndPort, EndpointState>> endpointStateMapSupplier = () -> new HashMap<InetAddressAndPort, EndpointState>() {{put(endpoint, localState);}};
 
-            Optional<Map.Entry<InetAddressAndPort, EndpointState>> entry = Gossiper.instance.endpointStateMap.entrySet()
```

Comment:
Nit: Now the import of `Optional` is not used anymore.

Reply:
ah right, thanks, always forget to check that ...